### PR TITLE
Fix problems

### PR DESCRIPTION
**Pull request by Rohanmond**
Optimized Wishlist and cartlist time complexity from o(n2) to o(n)
Fixed search bar
preview =>https://fashi-fy-react-hsuusduj5-rohanmond.vercel.app/

### CODE REVIEW

File: fashify/src/contexts/data-context.js [hunk 1]
```diff
@@ -19,11 +19,12 @@ import { useAuth } from "./auth-context";
 
 const DataContext = createContext();
 export const DataProvider = ({ children }) => {
-  const { token } = useAuth();
+  console.log("context is calling")
 
   const [state, dispatch] = useReducer(DataReducer, initialState);
   const [loader, setLoader] = useState(false);
   useEffect(() => {
+    console.log("use Effect called")
```
Comment (romabulani) [inline fashify/src/contexts/data-context.js]: You can remove these console logs

File: fashify/src/pages/ProductList/components/ProductCard/ProductCard.jsx [hunk 2]
```diff
@@ -13,15 +13,15 @@ import { ActionType } from "../../../../DataReducer/constants";
...
-  const [carted, setCart] = useState(false);
-  const [wished, setWished] = useState(false);
-  useEffect(() => {
-    state.wishlist.find((el) => el._id === _id) && setWished(true);
-    state.cartlist.find((el) => el._id === _id) && setCart(true);
-    console.log("cart", state.cartlist);
-  }, [state.wishlist, state.cartlist]);
+  console.log("product",product)
+  console.log("product card called")
+  // useEffect(() => {
```
Comment (romabulani) [inline fashify/src/pages/ProductList/components/ProductCard/ProductCard.jsx]: Here also, if not required

**Review by romabulani — APPROVED** [review]
- Liked that you have used keydown event for search
- Also, the naming of constants is in uppercase which is the good practice

File: fashify/src/contexts/data-context.js (hunk 1 above)
Comment (Rohanmond) [reply fashify/src/contexts/data-context.js]: thanks

**Review by shrey27 — APPROVED** [review]
Nice optimization. 
Code looks much cleaner and organised now.
PR is ready to merge.

**Pull request merged**